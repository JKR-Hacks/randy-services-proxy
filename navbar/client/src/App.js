/* eslint-disable prefer-destructuring */
/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import axios from 'axios';

import Navigation from './components/Navigation';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      teams: [],
      view: 'main',
    };
    this.renderView = this.renderView.bind(this);
  }

  componentDidMount() {
    axios.get('http://localhost:3007/espn/teamstandings', {
      method: 'GET',
      mode: 'no-cors',
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Content-Type': 'application/json',
      },
    })
      .then((res) => {
        const data = res.data;
        this.setState({
          teams: data,
        });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  // eslint-disable-next-line consistent-return
  renderView() {
    const { view, teams } = this.state;
    if (view === 'main') {
      return (
        <div>
          <Navigation teams={teams} />
        </div>
      );
    }
  }

  render() {
    return (
      this.renderView()
    );
  }
}
