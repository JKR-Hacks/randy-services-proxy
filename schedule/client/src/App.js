/* eslint-disable consistent-return */
import React, { Component } from 'react';
import axios from 'axios';

import Schedule from './components/Schedule';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      schedule: [],

      view: 'schedule',
    };
  }

  componentDidMount() {
    // fetch schedule data
    axios.get('http://localhost:3002/espn/schedules', {
      method: 'GET',
      mode: 'no-cors',
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Content-Type': 'application/json',
      },
    })
      .then((res) => {
        // eslint-disable-next-line prefer-destructuring
        const data = res.data;
        this.setState({
          schedule: data,
        });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  renderView() {
    const { schedule, view } = this.state;
    if (view === 'schedule') {
      return (
        <div id="fullschedule">
          <Schedule
            ramsSchedule={schedule}
          />
        </div>
      );
    }
    // if (view === 'feed') {
    //   return (
    //     <div id="feed">
    //       <Feed
    //         ramsFeed={ramsFeed}
    //       />
    //     </div>
    //   );
    // }
    // if (view === 'sidebarSchedule') {
    //   return (
    //     <div id="sidebarSchedule">
    //       <SidebarSchedule
    //         ramsSchedule={schedule}
    //       />
    //     </div>
    //   );
    // }
  }

  render() {
    return (
      <div>
        {this.renderView()}
      </div>
    );
  }
}

export default App;
