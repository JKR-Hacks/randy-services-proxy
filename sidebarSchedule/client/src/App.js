/* eslint-disable no-unused-vars */
/* eslint-disable consistent-return */
import React, { Component } from 'react';
import axios from 'axios';

import SidebarSchedule from './components/SidebarSchedule';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      schedule: [],
      view: 'main',
    };
    this.handleClick = this.handleClick.bind(this);
  }

  componentDidMount() {
    // fetch schedule data
    axios.get('http://localhost:3002/espn/schedules', {
      method: 'GET',
      mode: 'no-cors',
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Content-Type': 'application/json',
      },
    })
      .then((res) => {
        // eslint-disable-next-line prefer-destructuring
        const data = res.data;
        this.setState({
          schedule: data,
        });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  handleClick() {
    const { view } = this.state;
    this.setState({
      view: 'schedule',
    });
  }

  renderView() {
    const { schedule, view } = this.state;
    if (view === 'main') {
      return (
        <div id="SidebarSchedule">
          <SidebarSchedule
            ramsSchedule={schedule}
            handleClick={this.handleClick}
          />
        </div>
      );
    }
  }

  render() {
    return (
      <div>
        {this.renderView()}
      </div>
    );
  }
}

export default App;
