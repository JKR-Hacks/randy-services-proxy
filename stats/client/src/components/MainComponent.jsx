import React from 'react';

const MainComponent = ({ sample }) => (

  <div id="container" className="col-lg-3 alert alert-light container">
    <p className="text-dark font-weight-bold text-left">Team Stats</p>
    <div className="row">

      <div id="lefttop" className="col-sm border-right border-bottom border-secondary text-center">
        <p id="headingComp">Passing:</p>
        <br />
        <p className="h4 text-dark font-weight-bold">{sample.reduce((elem, i) => elem + i.Offense.Passing, 0) / sample.length}</p>
      </div>

      <div id="righttop" className="col-sm border-bottom border-secondary text-center">
        <p id="headingComp">Rushing:</p>
        <br />
        <p className="h4 text-dark font-weight-bold">{sample.reduce((elem, i) => elem + i.Offense.Rushing, 0) / sample.length}</p>
      </div>

    </div>

    <div className="row">

      <div id="leftbottom" className="col-sm border-right border-bottom border-secondary text-center">
        <p id="headingComp">Tackles:</p>
        <br />
        <p className="h4 text-dark font-weight-bold">{sample.reduce((elem, i) => elem + i.Defense.Tackles, 0) / sample.length}</p>
      </div>

      <div id="rightbottom" className="col-sm border-bottom border-secondary text-center">
        <p id="headingComp">Interceptions:</p>
        <br />
        <p className="h4 text-dark font-weight-bold">{sample.reduce((elem, i) => elem + i.Defense.Interceptions, 0) / sample.length}</p>
      </div>

    </div>
    <br />
    <a href="http://www.espn.com/nfl/statistics" target="_blank"><p className="text-center">All Team Rankings</p></a>

  </div>

);


export default MainComponent;
