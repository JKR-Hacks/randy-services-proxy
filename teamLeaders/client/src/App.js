/* eslint-disable react/jsx-no-bind */
/* eslint-disable import/extensions */
import React, { Component } from 'react';
import $ from 'jquery';
import Teamleaders from './components/Teamleaders.jsx';


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      sample: [],
      photos: [],
      side: 'Offense',
    };
  }

  componentDidMount() {
    $.ajax({
      method: 'GET',
      url: 'http://localhost:3003/stats',
      mode: 'no-cors',
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Content-Type': 'application/json',
      },
      error: (err) => {
        console.log(err, 'err');
      },
      success: (data) => {
        this.setState({
          sample: data,
        });
      },
    });
    $.ajax({
      method: 'GET',
      url: 'http://localhost:3003/photos',
      mode: 'no-cors',
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Content-Type': 'application/json',
      },
      dataType: 'json',
      error: (err) => {
        console.log(err, 'err');
      },
      success: (data) => {
        this.setState({
          photos: data,
        });
      },
    });
  }

  changeSide(e) {
    this.setState({
      side: e,
    });
  }

  render() {
    const { sample, photos, side } = this.state;
    return (
      <div>
        <Teamleaders
          sample={sample}
          photos={photos}
          changeSide={this.changeSide.bind(this)}
          side={side}
        />
      </div>
    );
  }
}

export default App;
