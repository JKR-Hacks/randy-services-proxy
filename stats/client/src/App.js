/* eslint-disable import/extensions */
import React, { Component } from 'react';
import $ from 'jquery';
import MainComponent from './components/MainComponent.jsx';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      sample: [],
    };
  }

  componentDidMount() {
    $.ajax({
      method: 'GET',
      url: 'http://localhost:3004/stats',
      mode: 'no-cors',
      headers: {
        "Access-Control-Allow-Origin": '*',
        'Content-Type': 'application/json',
      },
      error: (err) => {
        console.log(err, 'err');
      },
      success: (data) => {
        this.setState({
          sample: data,
        });
      },
    });
  }


  render() {
    const { sample } = this.state;
    return (
      <div>
        <MainComponent sample={sample} />
      </div>
    );
  }
}

export default App;
