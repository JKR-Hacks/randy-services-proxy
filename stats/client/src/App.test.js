/* eslint-disable no-unused-vars */
/* eslint-disable import/no-unresolved */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import { doesNotReject } from 'assert';
import App from './App';
import MainComponent from './components/MainComponent.jsx/';

describe('App component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});

it('Should update a state in a callback', () => {
  const wrapper = shallow(<App />);
  wrapper.setState({ sample: 'new' }, () => {
    expect(wrapper.state()).toEqual({ sample: 'new' });
  });
});

it('Should call componentDidMount', () => {
  const spy = jest.spyOn(App.prototype, 'componentDidMount');
  const wrapper = mount(<App />);
  wrapper.instance().componentDidMount();
  expect(spy).toHaveBeenCalled();
});

it('Should return a div element', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.type()).toEqual('div');
});
