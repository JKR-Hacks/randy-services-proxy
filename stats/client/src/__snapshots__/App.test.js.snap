// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div>
  <MainComponent
    sample={Array []}
  />
</div>
`;
